ori/volume/server/volume_create.go

```diff
@@ -16,261 +16,143 @@ package server
 
 import (
 	"context"
+	"errors"
 	"fmt"
-	"strings"
 
 	"github.com/go-logr/logr"
-	snapshotv1 "github.com/kubernetes-csi/external-snapshotter/client/v6/apis/volumesnapshot/v1"
-	volumev1alpha1 "github.com/onmetal/cephlet/ori/volume/api/v1alpha1"
-	"github.com/onmetal/cephlet/ori/volume/apiutils"
-	"github.com/onmetal/controller-utils/metautils"
-	storagev1alpha1 "github.com/onmetal/onmetal-api/api/storage/v1alpha1"
+	"github.com/onmetal/cephlet/pkg/api"
+	"github.com/onmetal/cephlet/pkg/limits"
+	"github.com/onmetal/cephlet/pkg/round"
+	"github.com/onmetal/cephlet/pkg/store"
 	ori "github.com/onmetal/onmetal-api/ori/apis/volume/v1alpha1"
 	onmetalimage "github.com/onmetal/onmetal-image"
-	"github.com/onmetal/onmetal-image/oci/image"
-	"github.com/onmetal/onmetal-image/oci/remote"
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/api/resource"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/utils/pointer"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func GetSanitizedImageName(image string) string {
-	image = strings.ReplaceAll(image, "/", "-")
-	image = strings.ReplaceAll(image, ":", "-")
-	return strings.ReplaceAll(image, "@", "-")
+type aggregatedImage struct {
+	image    *api.Image
+	snapshot *api.Snapshot
 }
 
-func getImageSize(ctx context.Context, imageName string) (*resource.Quantity, error) {
-	reg, err := remote.DockerRegistry(nil)
-	if err != nil {
-		return nil, fmt.Errorf("failed to initialize registry: %w", err)
-	}
-
-	img, err := reg.Resolve(ctx, imageName)
-	if err != nil {
-		return nil, fmt.Errorf("failed to resolve image ref in registry: %w", err)
-	}
-
-	layers, err := img.Layers(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get layers for image: %w", err)
-	}
+func (s *Server) getAggregatedImage(ctx context.Context, log logr.Logger, volume *ori.Volume) (*aggregatedImage, error) {
```

Comment:
If this is used to compute the configuration (i.e. required values for the internal API), this should be called `getCephVolumeConfig`. The return type should then be called `CephVolumeConfig`.